corpusreaders/src/main/java/edu/illinois/cs/cogcomp/nlp/corpusreaders/ontonotes/OntonotesNamedEntityReader.java

```diff
@@ -0,0 +1,247 @@
+     * parse the pen treebank parse file, producing an annotation covering the entire file.
+     * @param lines the data from the file, each line.
+     * @return the text annotation.
+     * @throws AnnotatorException
+     */
+    private XmlTextAnnotation nextAnnotation(String data,String docid) throws AnnotatorException {
+        
+        // we keep everything.
+        XmlDocumentProcessor xmlProcessor = new XmlDocumentProcessor(tagsWithText, tagsWithAtts, dropTags, true);
+        StatefulTokenizer st = new StatefulTokenizer();
+        TokenizerTextAnnotationBuilder taBuilder = new TokenizerTextAnnotationBuilder(st);
+        XmlTextAnnotationMaker xtam = new XmlTextAnnotationMaker(taBuilder, xmlProcessor);
+ 
+        // read the file and create the annotation.
+        XmlTextAnnotation xta = xtam.createTextAnnotation(data, "OntoNotes 5.0", docid);
+        TextAnnotation ta = xta.getTextAnnotation();
+        List<SpanInfo> fudge = xta.getXmlMarkup();
+ 
+        // create the named entity vi
+        View nerView = new SpanLabelView(ViewNames.NER_ONTONOTES, ta);
+        for (SpanInfo si : fudge) {
+            if ("enamex".equalsIgnoreCase(si.label)) {
+ 
+                IntPair charOffsets = si.spanOffsets;
+                String neLabel = si.attributes.get("type").getFirst();
+                int cleanTextCharStart = xta.getXmlSt().computeModifiedOffsetFromOriginal(charOffsets.getFirst());
+                int cleanTextCharEnd = xta.getXmlSt().computeModifiedOffsetFromOriginal(charOffsets.getSecond());
+                int cleanTextNeTokStart = ta.getTokenIdFromCharacterOffset(cleanTextCharStart);
+                int cleanTextNeTokEnd = ta.getTokenIdFromCharacterOffset(cleanTextCharEnd-1); // StringTransformation returns one-past-the-end index; TextAnnotation maps at-the-end index
+                if (cleanTextNeTokStart == -1 || cleanTextNeTokEnd == -1) {
+                    for (Constituent c : nerView.getConstituents()) {
+                        System.err.println(c);
+                    }
+                    System.err.println("Something wonky in \""+docid+"\", at "+charOffsets+", "+cleanTextCharStart+" - "+cleanTextCharEnd
+                        +" = "+ta.text.substring(cleanTextCharStart, cleanTextCharEnd));
+                } else {
+                    Constituent neCon = new Constituent(neLabel, nerView.getViewName(), ta, cleanTextNeTokStart, cleanTextNeTokEnd + 1); //constituent token indexing uses one-past-the-end
+                    nerView.addConstituent(neCon);
+                }
+            }
+        }
+        ta.addView(ViewNames.NER_ONTONOTES, nerView);
+        return xta;
+
+    }
+
+    /**
+     * TODO: generate a human-readable report of annotations read from the source file (plus whatever
+     * other relevant statistics the user should know about).
+     */
+    public String generateReport() {
+        throw new UnsupportedOperationException("ERROR: generateReport() Not yet implemented.");
```

Comment:
ideally, please track counts of entity types and in this method write a human-readable string (simple tab-separated table?) reporting them.  Also, if it is possible that the reader will fail, report e.g. the number of files that could not be read, and list those files; or, if failure is localized, the number of times an error was generated. 
IDEALLY we want to know how many annotations, and what types, were in the original files, so that when we report performance, we account for anything we couldn't read.